### PR TITLE
Build docker image on arm64

### PR DESCRIPTION
**Pull request by Yc-Chen**
## Checklist

- [ ] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [ ] Tests for the changes have been added
- [ ] CI has been passed. (GitHub actions all turns green)
- [ ] CLA has been signed

## Description

Added changes to build docker image on a raspberrypi installed with 64bit RaspiOS. As discussed in 

- https://github.com/wechaty/wechaty/issues/2248

### CODE REVIEW

**Comment by Yc-Chen** [comment]
I actually haven't been able to run the docker image. But I followed the same steps as the 'bot.js' example:
https://wechaty.js.org/docs/tutorials/docker/#run

```
import { Wechaty } from 'wechaty'

Wechaty.instance() // Singleton
  .on('scan', (qrcode, status) => console.log(`Scan QrCode to login: ${status}\$
  .on('login',       user      => console.log(`User ${user} logined`))
  .on('message',  message      => console.log(`Message: ${message}`))
  .start()
```

Then I got the same error on mac so it is unlikely because of my changes:

```
Starting Docker Container for Wechaty v1.11.1 with Node.js v16.13.0 ...
Working directory: /bot
Executing node bot.js
file:///bot/bot.js:1
import { Wechaty } from 'wechaty'
         ^^^^^^^
SyntaxError: The requested module 'wechaty' does not provide an export named 'Wechaty'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:124:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:181:5)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:281:24)
    at async loadESM (node:internal/process/esm_loader:88:5)
    at async handleMainPromise (node:internal/modules/run_main:65:12)
ERROR: Bot exited with code 1
```

**Comment by huan** [comment]
Our docs website is outdated with the latest Wechaty v1.x.

I have sent a PR to fix them, and you can test your docker image with the below code which is adapted with v1.x:

```diff
- import { Wechaty } from 'wechaty'
+ import { WechatyBuilder } from 'wechaty'

- Wechaty.instance() // Singleton
+ WechatyBuilder.build()
  .on('scan', (qrcode, status) => console.log(`Scan QrCode to login: ${status}\$
  .on('login',       user      => console.log(`User ${user} logined`))
  .on('message',  message      => console.log(`Message: ${message}`))
  .start()
```

When you have confirmed that it can work on Raspberry, then we can merge this great PR.

Thank you very much.

File: Dockerfile [hunk 1]
```diff
@@ -51,10 +51,10 @@ RUN  wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-ke
   && sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list' \
```
Comment (huan) [inline Dockerfile:51]: I think if we switch to using the `chromium`, then the chrome apt source can be removed.

**Comment by Yc-Chen** [comment]
Thanks for the change!

I still get error running the docker image, and they are different on different platforms:

On mac:
```
Starting Docker Container for Wechaty v1.11.1 with Node.js v16.13.0 ...
Working directory: /bot
Executing node bot.js

wechaty-puppet-service: WECHATY_PUPPET_SERVICE_TOKEN not found
(save token to WECHATY_PUPPET_SERVICE_TOKEN env var or pass it to puppet options is required.).

To learn how to get Wechaty Puppet Service Token,
please visit <https://wechaty.js.org/docs/puppet-services/>
to see our Wechaty Puppet Service Providers.

node:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

GError: wechaty-puppet-service: WECHATY_PUPPET_SERVICE_TOKEN not found
    at Object.WECHATY_PUPPET_SERVICE_TOKEN (file:///wechaty/node_modules/wechaty-puppet-service/dist/esm/src/env-vars.js:34:11)
    at new PuppetService (file:///wechaty/node_modules/wechaty-puppet-service/dist/esm/src/client/puppet-service.js:67:28)
    at Function.resolve (file:///wechaty/dist/esm/src/puppet-management/puppet-manager.js:65:32)
    at async WechatyImpl.__initPuppetInstance (file:///wechaty/dist/esm/src/wechaty-mixins/puppet-mixin.js:108:36)
    at async WechatyImpl.start (file:///wechaty/dist/esm/src/wechaty-mixins/puppet-mixin.js:44:17)
    at async WechatyImpl.start (file:///wechaty/dist/esm/src/wechaty-mixins/login-mixin.js:85:13)
    at async WechatyImpl.start (file:///wechaty/dist/esm/src/wechaty-mixins/io-mixin.js:18:13)
    at async WechatyImpl.start (file:///wechaty/node_modules/state-switch/dist/esm/src/service-ctl/service-ctl.js:55:21)
Emitted 'error' event on WechatyImpl instance at:
    at WechatyImpl.emit (file:///wechaty/dist/esm/src/wechaty-mixins/gerror-mixin.js:45:26)
    at WechatyImpl.start (file:///wechaty/node_modules/state-switch/dist/esm/src/service-ctl/service-ctl.js:70:22) {
  code: 2,
  details: 'Error: wechaty-puppet-service: WECHATY_PUPPET_SERVICE_TOKEN not found\n' +
    '    at Object.WECHATY_PUPPET_SERVICE_TOKEN (file:///wechaty/node_modules/wechaty-puppet-service/dist/esm/src/env-vars.js:34:11)\n' +
    '    at new PuppetService (file:///wechaty/node_modules/wechaty-puppet-service/dist/esm/src/client/puppet-service.js:67:28)\n' +
    '    at Function.resolve (file:///wechaty/dist/esm/src/puppet-management/puppet-manager.js:65:32)\n' +
    '    at async WechatyImpl.__initPuppetInstance (file:///wechaty/dist/esm/src/wechaty-mixins/puppet-mixin.js:108:36)\n' +
    '    at async WechatyImpl.start (file:///wechaty/dist/esm/src/wechaty-mixins/puppet-mixin.js:44:17)\n' +
    '    at async WechatyImpl.start (file:///wechaty/dist/esm/src/wechaty-mixins/login-mixin.js:85:13)\n' +
    '    at async WechatyImpl.start (file:///wechaty/dist/esm/src/wechaty-mixins/io-mixin.js:18:13)\n' +
    '    at async WechatyImpl.start (file:///wechaty/node_modules/state-switch/dist/esm/src/service-ctl/service-ctl.js:55:21)'
}
ERROR: Bot exited with code 1
```

On raspberrypi:
```
Starting Docker Container for Wechaty v1.11.1 with Node.js v16.13.0 ...
Working directory: /bot
Please make sure you had installed all the NPM modules which is depended on your bot script.
Executing node bot.js
node:internal/process/esm_loader:94
    internalBinding('errors').triggerUncaughtException(
                              ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/bot/dist/esm/src/mods/mod.js' imported from /bot/bot.js
Did you mean to import wechaty/dist/cjs/src/mods/mod.js?
    at new NodeError (node:internal/errors:371:5)
    at finalizeResolution (node:internal/modules/esm/resolve:405:11)
    at moduleResolve (node:internal/modules/esm/resolve:932:10)
    at defaultResolve (node:internal/modules/esm/resolve:1044:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:422:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:222:40)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
ERROR: Bot exited with code 1
```

**Comment by huan** [comment]
By default, wechaty is using the `wechaty-puppet-service` provider. Puppet Service needs a TOKEN before it works.

You can create a token easily by following http://wechaty.js.org/docs/puppet-services/diy ,

Or specifying a [free puppet](http://wechaty.js.org/docs/puppet-providers) like `wechaty-puppet-wechat`:

```sh
export WECHATY_PUPPET=wechaty-puppet-wechat
```

Pass it with the `options` in the code:

```ts
Const wechaty = WechatyBuilder.build({
  puppet: 'wechaty-puppet-wechat',
}) 
```

**Comment by huan** [comment]
I have added multi-platform support to the scripts (build & ci).

Please take a look at them and then we can try to merge this PR to see if it will work as expected.

**Comment by Yc-Chen** [comment]
Now I wondering if I miss something in my local build environment because I _can_ run the image from dockerhub but I still get errors with the image built locally. in both mac and raspberrypi, i did:

```
docker build -t wechaty .
// cd (a clean folder with only bot.js)
docker run -ti --rm --volume="$(pwd)":/bot --env WECHATY_PUPPET=wechaty-puppet-wechat wechaty bot.js
```

Then I got error:
```
node:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

file:///wechaty/node_modules/wechaty-puppet-wechat/dist/esm/src/bridge.js:27
import { wrapAsyncError } from 'wechaty-puppet/helpers';
         ^^^^^^^^^^^^^^
SyntaxError: The requested module 'wechaty-puppet/helpers' does not provide an export named 'wrapAsyncError'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:124:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:181:5)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:281:24)
    at async resolvePuppetName (file:///wechaty/node_modules/wechaty-puppet/dist/esm/src/puppet/puppet-resolver.js:62:24)
    at async Module.resolvePuppet (file:///wechaty/node_modules/wechaty-puppet/dist/esm/src/puppet/puppet-resolver.js:34:22)
    at async WechatyImpl.__initPuppetInstance (file:///wechaty/dist/esm/src/wechaty-mixins/puppet-mixin.js:106:36)
    at async WechatyImpl.start (file:///wechaty/dist/esm/src/wechaty-mixins/puppet-mixin.js:43:17)
    at async WechatyImpl.start (file:///wechaty/dist/esm/src/wechaty-mixins/login-mixin.js:84:13)
    at async WechatyImpl.start (file:///wechaty/dist/esm/src/wechaty-mixins/io-mixin.js:18:13)
    at async WechatyImpl.start (file:///wechaty/node_modules/state-switch/dist/esm/src/service-ctl/service-ctl.js:55:21)
Emitted 'error' event on WechatyImpl instance at:
    at WechatyImpl.emit (file:///wechaty/dist/esm/src/wechaty-mixins/gerror-mixin.js:45:26)
    at WechatyImpl.start (file:///wechaty/node_modules/state-switch/dist/esm/src/service-ctl/service-ctl.js:70:22)
```

I uploaded the 2 images to onedrive here: https://1drv.ms/f/s!ApWqhOVaNIvhhBO4HzGSKzpe6Qvg

If this is only my environment problem and the rest looks ok, please feel free to merge it. If you would like, I'm also happy to schedule a session with you to debug it further. @huan

**Comment by huan** [comment]
Thanks for the feedback, and great to know that you can run the image from DockerHub! (it was published when I was testing locally)

I'm sorry for the problem you have experienced, I can confirm it is a bug which is introduced by the breaking change in the latest `wechaty@1.11`:

- the `wechaty-puppet/helpers` import entry has been deprecated/removed in the `wechaty-puppet@1.11`
- the last version that contains the `helpers` is `wechaty-puppet@1.10`

I have submitted a commit to fix this issue and it will be published as `wechaty-puppet-wechat@1.10.3`.

And as we all agreed that we can merge this PR, so I'll do it as soon as the `wechaty-puppet-wechat@1.10.3` is published successfully, and let's keep our eyes on the pipeline and continue improving it together.

Cheers!

**Review by huan — APPROVED** [review]
LGTM

**Comment by huan** [comment]
Thank you very much for your contribution!

## You are welcome to join [Wechaty Contributor Program](https://wechaty.js.org/docs/contributor-program/)

### 1. Join Wechaty Organization

> You've invited Yicong Chen to Wechaty! They'll be receiving an email shortly. They can also visit https://github.com/wechaty to accept the invitation.

I have invited you to join our Wechaty GitHub Organization, please accept it by following the above message. (See also: [wechaty/PMC#16](https://github.com/wechaty/PMC/issues/16))

### 2. Update Your Wechaty Contributor Profile

1. Please open [Contributor Hall of Fame](https://docs.google.com/spreadsheets/d/1XcDoIczyIclqXP1p90Sz7S0n4Q22xjFzJmjPFlU2g1E/edit#gid=0) and add yourself to the end of the list, so that other contributors will know you better!
1. Please add yourself to our Website [Contributors](https://wechaty.js.org/contributors/) by [creating a PR](https://github.com/wechaty/wechaty.js.org/tree/master/jekyll/_contributors) and refer to this PR link as well.

### 3. Join The Contributor Only WeChat Room

We also have a WeChat room for contributors only which can discuss Wechaty at a deeper level, you are welcome to join and if you are interested.

Please add @lijiarui wechat: _ruirui_0914_ and send her this pr link. She will invite you into Wechaty Contributor Room

Cheers!

**Pull request merged**